Source/WebCore/style/AnchorPositionEvaluator.cpp

```diff
@@ -1085,14 +1086,19 @@ CSSPropertyID AnchorPositionEvaluator::resolvePositionTryFallbackProperty(CSSPro
     return propertyID;
 }
 
-bool AnchorPositionEvaluator::overflowsContainingBlock(const RenderBox& anchoredBox)
+bool AnchorPositionEvaluator::overflowsInsetModifiedContainingBlock(const RenderBox& anchoredBox)
 {
-    auto containingBlockRect = anchoredBox.containingBlock()->contentBoxRect();
+    if (!anchoredBox.isOutOfFlowPositioned())
+        return false;
+
+    auto inlineConstraints = PositionedLayoutConstraints { anchoredBox, LogicalBoxAxis::Inline };
+    auto blockConstraints = PositionedLayoutConstraints { anchoredBox, LogicalBoxAxis::Block };
 
-    auto marginRect = anchoredBox.marginBoxRect();
-    marginRect.moveBy(anchoredBox.location());
+    auto anchorInlineSize = anchoredBox.logicalWidth() + anchoredBox.marginStart() + anchoredBox.marginEnd();
+    auto anchorBlockSize = anchoredBox.logicalHeight() + anchoredBox.marginBefore() + anchoredBox.marginAfter();
 
-    return !containingBlockRect.contains(marginRect);
+    return inlineConstraints.insetModifiedContainingBlockSize() < anchorInlineSize
+        || blockConstraints.insetModifiedContainingBlockSize() < anchorBlockSize;
```

Comment:
I think it would be better to do one axis and then the other, short-circuiting the PositionedLayoutConstraints constructor when it's not needed. That constructor does a lot of work.